cmd/metal-api/internal/fsm/fsm_test.go

```diff
@@ -0,0 +1,315 @@
+package fsm
+
+import (
+	"testing"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+	"github.com/metal-stack/metal-api/cmd/metal-api/internal/metal"
+	"go.uber.org/zap/zaptest"
+)
+
+func TestHandleProvisioningEvent(t *testing.T) {
+	now := time.Now()
+	lastEventTime := now.Add(-time.Minute * 4)
+	exceedThresholdTime := now.Add(-time.Minute * 10)
+	tests := []struct {
+		event     *metal.ProvisioningEvent
+		container *metal.ProvisioningEventContainer
+		name      string
+		wantErr   error
```

Comment:
There are no tests that have an error. Are there any interesting failure scenarios to cover?